db.go

```diff
@@ -8,7 +8,7 @@ import (
 
 // DB is the interface for bundling all database operations.
 type DB interface {
-	SaveURL(url string, keybuffer <-chan []byte) (string, error)
+	SaveURL(url string, key []byte) (string, error)
```

Comment:
What if the user-provided key already exists in the DB? We probably don't want to let users overwrite each other's keys.

Reply:
i kept the key-collision check in there, but i didn't write anything to alert the user.